### PR TITLE
Add the new animations

### PR DESCRIPTION
**Pull request by bulyonov**
This PR changes the way how the components are animated:

- removes d3-transition;
- does animation using react;
- adds a possibility for minor tweaks of the animation

These changes are made for `XYPlot`, `RadialChart` and `Treemap` (and all their subcomponents)

This issue addresses #55, #91, #114 and #123

### CODE REVIEW

File: src/lib/plot/axis/axis.js [hunk 1]
```diff
@@ -61,7 +70,7 @@ const defaultProps = {
   orientation: BOTTOM
 };
 
-class Axis extends PureRenderComponent {
+class Axis extends React.Component {
```
Comment (chrisirhc) [inline src/lib/plot/axis/axis.js]: How come this needs to be changed from PureRenderComponent?

File: src/lib/animation.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      interpolator: null
+    };
+    this._renderChildren = this._renderChildren.bind(this);
+  }
+
+  componentWillReceiveProps(props) {
+    const {data} = this.state;
+    const newData = this._extractPropsFromChild(props);
+    this.setState({
+      interpolator: interpolate(data, newData)
```
Comment (chrisirhc) [inline src/lib/animation.js]: Is this state to cache the `interpolator`?

File: src/lib/plot/axis/axis.js (hunk 1 above)
Comment (bulyonov) [inline src/lib/plot/axis/axis.js]: @chrisirhc You are right. It is still in progress, testing this out.

File: src/lib/animation.js (hunk 2 above)
Comment (bulyonov) [inline src/lib/animation.js]: Didn't know that, thanks!

File: src/lib/animation.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        _animation: Math.random()
+      }
+    );
+  }
+
+  render() {
+    const defaultStyle = {i: 0};
+    const style = {i: spring(1)};
+    return (
+      <Motion {...{defaultStyle, style, key: Math.random()}}>
```
Comment (chrisirhc) [inline src/lib/animation.js]: What's `key: Math.random()` for? If it's a required hack, there should be a comment here as well.

Comment (bulyonov) [inline src/lib/animation.js]: The spring is animated only at the time when motion renders. This means that it requires to be reinitialized somehow to make the next animation.

File: src/lib/plot/series/abstract-series.js [hunk 4]
```diff
@@ -104,47 +100,50 @@ export default class AbstractSeries extends PureRenderComponent {
    * Mouse over handler for all series.
    * @param {boolean} useValue Use value handler if true.
    * @param {Object} d Value object
+   * @param {Object} event Event.
    * @private
    */
-  _mouseOverHandler(useValue, d) {
+  _mouseOverHandler(useValue, d, event) {
```
Comment (chrisirhc) [inline src/lib/plot/series/abstract-series.js]: How is `event` passed to this?

Comment (bulyonov) [inline src/lib/plot/series/abstract-series.js]: @chrisirhc Event is passed from inside of each of the series. I made the change to make it more transparent.

**Pull request merged**